src/main/java/org/cactoos/codec/Base64ToInput.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.codec;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Base64;
+import org.cactoos.Input;
+import org.cactoos.text.BytesAsText;
+import org.cactoos.text.FormattedText;
+import org.cactoos.text.UncheckedText;
+
+/**
+ * Base64 to Input.
+ * <p>
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Mehmet Yildirim (memoyil@gmail.com)
+ * @version $Id$
+ * @since 0.12
+ */
+public final class Base64ToInput implements Input {
+
+    /**
+     * All legal Base64 chars.
+     */
+    private static final String BASE64CHARS =
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
+
+    /**
+     * Original input.
+     */
+    private final Input origin;
+
+    /**
+     * Ctor.
+     *
+     * @param input Original
+     */
+    public Base64ToInput(final Input input) {
+        this.origin = input;
+    }
+
+    @Override
+    public InputStream stream() throws IOException {
+        final InputStream stream = this.origin.stream();
+        final ByteArrayOutputStream out = new ByteArrayOutputStream();
+        int read = stream.read();
+        while (read != -1) {
+            out.write(read);
+            read = stream.read();
+        }
+        final byte[] bytes = out.toByteArray();
+        final byte[] illegal = Base64ToInput.getIllegalCharacters(bytes);
+        if (illegal.length > 0) {
+            throw new DecodingException(
+                new FormattedText(
+                    "Illegal character in Base64 encoded data. %s",
+                    1, new UncheckedText(new BytesAsText(illegal)).asString()
+                ).asString()
+            );
+        }
+        return new ByteArrayInputStream(Base64.getDecoder().decode(bytes));
+    }
+
+    /**
+     * Get the byte array for non-Base64 characters.
+     * @param bytes Byte Array
+     * @return An array of the found non-Base64 characters.
+     * @throws IOException If fails
+     */
+    private static byte[] getIllegalCharacters(final byte[] bytes) throws
+        IOException {
+        final ByteArrayOutputStream output = new ByteArrayOutputStream();
+        for (int pos = 0; pos < bytes.length; ++pos) {
+            if (BASE64CHARS.indexOf(bytes[pos]) < 0) {
```

Comment:
If we done like that we could lose information the illegal chars on DecodingException when exists. If we dont want to show illegal chars that the user gave us, we could use >0 . What should we do ?